docs/azure.md

```diff
@@ -266,6 +269,18 @@ azure {
 ```
 :::
 
+### Task packing on nodes
+
+Each node is given a number of task slots, which are the number of tasks that can be run concurrently on the node. The number of task slots is determined by the number of CPUs for the selected virtual machine. Nextflow will assign each process a number of task slots equal to a percentage of the total resources available on the node, based on the `cpus`, `memory`, and `disk` directives.
+
+For example, if using a `Standard_D4d_v5` machine with 4 vCPUs, 16 GB of memory, and a 150 GB local disk. If a process has the directives `cpus 2`, `memory 8.GB`, or `disk 75.GB`, it will be assigned one task slot and two tasks will concurrently run on the node. If the process has `cpus 4`, `memory 2.GB` or `disk 150.GB`, it will be assigned two task slots, and only one task will concurrently run on the node.
+
+A node may become overprovisioned if the tasks are using more than their fraction of total resources. For example, in the above example, if a process has the `cpus` directive set to `2`, it will be assigned one task slot and two tasks will concurrently run on the node. If the process uses more than 8 GB of memory or 75 GB of disk space, the node might become overprovisioned, and performance might degrade or the task will fail.
+
+:::{warning}
+The `cpus` directive is used to determine the number of task slots, not the number of cores.
```

Comment:
Are you saying that the number of task slots is determined solely by the requested vs available CPUs? If so then this note could be more clear